examples/map/rotate_custom_affine_transform_opencv.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Using a Custom Affine Transform in the Sunpy Map rotate() function
+# #### (with OpenCV)
+# ----------
+# How to construct a custom affine transform for `map.rotate`.
+# Requires OpenCV (cv2) Python library.
+
+import sunpy.data.sample
+import sunpy.map
+import numbers
+import numpy as np
+import cv2
+from sunpy.image.transform import _calculate_shift
+
+#############################
+# Rotating a map in sunpy (via `map.rotate()`) has a choice between three libraries:
+# `scipy`, `skimage`, and `cv2` (OpenCV).
+# However, the `method=` argument in `sunpy.map.rotate` can accept a custom function designed
+# to use an external library. Here, we illustrate this process by defining a function
+# with the OpenCV library; this will be identical to the built-in `cv2` option.
+
+# First, our custom method must have a similar function call to
+# `sunpy.image.transform.affine_transform` and return the same output,
+# namely the rotated, scaled, and transformed image.
+# The required arguments are:
+
+# Parameters
```

Comment:
I don't think this should be documented in the example as it makes the comment quite long. Can you link to the equivalent bit in the docs for `rotate` instead?